nomad/state/schema.go

```diff
@@ -254,6 +254,15 @@ func jobTableSchema() *memdb.TableSchema {
 					Field: "NodePool",
 				},
 			},
+			// ModifyIndex allows sorting by last-changed
+			"modify_index": {
```

Comment:
ditto here. should I split out to be able to backport?

Reply:
Same, fine to leave in